### PR TITLE
setup react router

### PR DESCRIPTION
**Pull request by semy-tuyi**
##  What does this PR do?

- set up react-router

##  Description of Task to be completed?

- setting react router for accessing different components through different routes

## How should this be manually tested?

- Clone the repo.

- checkout to a branch called ` ch-setup-react-router-#182201670`

- Run npm install to install modules.

- Run `npm run start` to run an app.
## What are the relevant pivotal tracker stories?
 - https://www.pivotaltracker.com/n/projects/2569105/stories/182201670

### CODE REVIEW

**Review by uwambajeddy — CHANGES_REQUESTED** [review]
`components folder` should have included `child components` of the `base components`  and also `view folder` should have contained `base components`

**Review by uwambajeddy — APPROVED** [review]
LGTM

**Review by alaintresor — APPROVED** [review]

File: coverage/clover.xml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (joshwambere) [inline coverage/clover.xml]: why did you push coverage folder on the repo

File: src/routes/index.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import React from "react";
```
Comment (joshwambere) [inline src/routes/index.js]: this All routes file might end up being a gigantic ones you might want to try a different approach

**Review by joshwambere — CHANGES_REQUESTED** [review]

File: package.json [hunk 3]
```diff
@@ -21,16 +21,18 @@
     "i": "^0.3.7",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
-    "react-scripts": "^5.0.1",
-    "react-test-renderer": "^18.2.0"
```
Comment (joshwambere) [inline package.json:31]: why did you remove this dependency

**Review by uwambajeddy — CHANGES_REQUESTED** [review]
you forgot to ignore  coverage folder

**Comment by semy-tuyi** [comment]
Screen Shot 2022-08-02 at 06 44 41
Screen Shot 2022-08-02 at 06 42 02
any suggestion of how I can remove that coverage folder?

File: src/routes/index.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+import Login from '../views/Login';
+import About from '../views/About';
+import SignUp from '../views/Signup';
+
+
+export default function Router() {
+    let element = useRoutes([
+        {
+            path: "/",
+            element: <Nav />,
```
Comment (joshwambere) [inline src/routes/index.js]: nav should not be in here since is not a route itself

File: src/routes/index.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+            children: [
```
Comment (joshwambere) [inline src/routes/index.js]: those are routes by themselves should be independent

File: src/routes/index.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+import Nav from '../components/NavDummy';
+import Home from '../views/Home';
+import Login from '../views/Login';
+import About from '../views/About';
+import SignUp from '../views/Signup';
+
+
+export default function Router() {
+    let element = useRoutes([
+        {
```
Comment (joshwambere) [inline src/routes/index.js]: you should have a root route `/` associated to `home`

File: src/App.js [hunk 7]
```diff
@@ -0,0 +1,12 @@
+import React from 'react'
```
Comment (joshwambere) [inline src/App.js:1]: if this file what is does is simply importing routes i think you can merge it with indes.js a

File: src/App.js [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+import Router from './routes'
+
+const App = () => {
+    return (
+        <>
+            <Router />
```
Comment (joshwambere) [inline src/App.js:9]: nav should be in here

**Review by joshwambere — CHANGES_REQUESTED** [review]

**Review by uwambajeddy — APPROVED** [review]

**Review by Placide123 — APPROVED** [review]

**Review by alaintresor — APPROVED** [review]

File: src/App.js [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+import Router from './routes'
+import Nav from './components/NavDummy';
+import Counter from './components/Counter'
+
+
+const App = () => {
+    return (
+        <>
+            <Nav />
+            <Counter />
```
Comment (joshwambere) [inline src/App.js]: Please move this counter into home component

**Review by joshwambere — CHANGES_REQUESTED** [review]

**Review by joshwambere — APPROVED** [review]

**Review by joshwambere — APPROVED** [review]

**Pull request merged**